src/inthandler.S

```diff
@@ -177,13 +177,21 @@ interrupt:
 	nop
 
 	/* handle reset */
-	addiu a0, sp, 32
 	jal __onResetException
-	nop
-
-	j end_interrupt
-	nop
+	addiu a0, sp, 32
 
+	# There is no way to ack the pre-NMI interrupt, so it will
+	# stay pending in CR. Let's disable it in SR to avoid
+	# looping here. If another unrelated interrupt triggers, 
+	# CR will still have 0x1000 set, but __onResetException will
+	# do nothing after the first call.
+	li t0, ~0x1000
+	lw t1, STACK_SR(sp)
+	and t1, t0
+	sw t1, STACK_SR(sp)
+
+	# Reload cause register and test for other interrupts
```

Comment:
Why we need a reload now? Honestly I didn't get the later ones as well.

Reply:
It's part of the commit https://github.com/DragonMinded/libdragon/pull/294/commits/a17748a40d177e30aa3eef8af899fe390c544fef. 

There is an explanation there. Basically, on trunk we currently process one interrupt per exception. That is, when an exception fires, we look for which interrupt has triggered; as soon as we find one interrupt, we call the handler, and then exit the exception. If another interrupt was pending, a new exception will immediately retrigger.

This is acceptable for the rare case of two semi-simultaneous interrupts, but once we activate the pre-NMI interrrupt it doesn't work anymore: in fact, pre-NMI cannot be acked and will always trigger. This means that, if the exception handler finds pre-NMI set, process it, and then exit, it will enter a deadlock where an exception is continuously generated.

So we change the exception handler to process all interrupts that it finds set. To do so, after calling a handler, we need to reload the COP0 cause register as the register in which it was stored might have been reused by the handler itself.

Reply:
BTW this is the most scary part of the whole PR, make sure it makes sense :)